pkg/private/zip/zip.bzl

```diff
@@ -138,7 +138,10 @@ limited to a granularity of 2 seconds.""",
         ),
 
         # Common attributes
-        "out": attr.output(mandatory = True),
+        "out": attr.output(
+            doc = """output file name. Default: name + ".zip".""",
+            mandatory = True,
```

Comment:
A `default` declaration seems conflict with the mandatory flag?

Reply:
It is not an actual default.  The default is done in the macro wrapper.
We do this kind of thing all the time because stardoc does not have the ability to merge the declaration from the rule with the wrapper you use around it.
Lots of discussion about that in https://github.com/bazelbuild/stardoc/issues/27

Reply:
I meant if it's a mandatory field, I assume it will throw error when `out` is not given, and default value is meaningless. It's confusing when doc says there'll be a default value to a mandatory field.

Reply:
Yes. It is confusing, but I don't have a good way around it.
The field *is* mandatory at the rule level, but is optional (with a default) at the wrapper macro.  This is a common idiom in starlark rules.